src/routes/profile.tsx

```diff
@@ -171,9 +178,134 @@ router.post(
   },
 );
 
-// Show the customavatar edit page
+// Show the user's avatar homepage
+router.get(
+  '/:profileUsername/avatarhome',
+  checkProfileOwnership,
+  async (req, res) => {
+    const user = await User.findOne({
+      usernameLower: req.params.profileUsername.toLowerCase(),
+    });
+
+    // TODO-kev: Remove below once library is fully updated. Remains for testing purposes
+    const approvedAvatarSetsQuery = await AvatarRequest.find({
+      forUsername: user.usernameLower,
+    });
+
+    let approvedAvatarSets = [];
+
+    approvedAvatarSetsQuery.forEach((avatarSet) => {
+      const approvedAvatarSet = {
+        resLink: avatarSet.resLink,
+        spyLink: avatarSet.spyLink,
+      };
+      approvedAvatarSets.push(approvedAvatarSet);
+    });
+
+    const currentPatreon = true;
+    const avatarHomeReact = renderToString(<AvatarHome />);
+
+    res.render('profile/avatarhome', {
+      username: user.username,
+      avatarImgRes: user.avatarImgRes,
+      avatarImgSpy: user.avatarImgSpy,
+      approvedAvatarSets,
+      currentPatreon,
+      avatarHomeReact,
+    });
+  },
+);
+
+// Get a users avatar library links
+router.get(
+  '/:profileUsername/avatar/avatarinfo',
+  checkProfileOwnership,
+  async (req, res) => {
+    const user = await User.findOne({
+      usernameLower: req.user.usernameLower,
+    });
+
+    const result = {
+      currentResLink: user.avatarImgRes,
+      currentSpyLink: user.avatarImgSpy,
+      avatarLibrary: await s3Agent.getUsersAvatarLibraryLinks(
+        user.usernameLower,
```

Comment:
Let's change this to provide in the avatar library with avatar sets and "transform" them into links so the agent itself doesn't need to access the user model.